vnext/Microsoft.ReactNative/ReactNativeHost.cpp

```diff
@@ -30,13 +33,85 @@ using namespace xaml::Controls;
 
 namespace winrt::Microsoft::ReactNative::implementation {
 
+class FuseboxHostTargetDelegate : public facebook::react::jsinspector_modern::HostTargetDelegate,
+                                  public std::enable_shared_from_this<FuseboxHostTargetDelegate> {
+ public:
+  FuseboxHostTargetDelegate(ReactNativeHost *reactNativeHost) : m_reactNativeHost(reactNativeHost) {}
+
+  void onReload(facebook::react::jsinspector_modern::HostTargetDelegate::PageReloadRequest const &request) override {
+    m_reactNativeHost->ReloadInstance();
+  }
+
+#ifdef HAS_FUSEBOX_PAUSED_OVERLAY // Remove after syncing past https://github.com/facebook/react-native/pull/44078
+  void onSetPausedInDebuggerMessage(
+      facebook::react::jsinspector_modern::HostTargetDelegate::OverlaySetPausedInDebuggerMessageRequest const &request)
+      override {
+    const auto instanceSettings = m_reactNativeHost->InstanceSettings();
+    if (instanceSettings) {
+      if (request.message.has_value()) {
+        ::Microsoft::ReactNative::DebuggerNotifications::OnShowDebuggerPausedOverlay(
+            instanceSettings.Notifications(), request.message.value(), [weakThis = weak_from_this()]() {
+              if (auto strongThis = weakThis.lock()) {
+                strongThis->m_reactNativeHost->OnDebuggerResume();
+              }
+            });
+      } else {
+        ::Microsoft::ReactNative::DebuggerNotifications::OnHideDebuggerPausedOverlay(instanceSettings.Notifications());
+      }
+    }
+  }
+#endif
+
+ private:
+  ReactNativeHost *m_reactNativeHost;
+};
+
 ReactNativeHost::ReactNativeHost() noexcept : m_reactHost{Mso::React::MakeReactHost()} {
 #if _DEBUG
   facebook::react::InitializeLogging([](facebook::react::RCTLogLevel /*logLevel*/, const char *message) {
     std::string str = std::string("ReactNative:") + message;
     OutputDebugStringA(str.c_str());
   });
 #endif
+
+  auto &inspectorFlags = facebook::react::jsinspector_modern::InspectorFlags::getInstance();
```

Comment:
This should all be lazy init'd on the first reload, otherwise you have to set the feature flag before calling the ReactNativeHost ctor, rather than just before attaching the ReactNativeHost to the ReactRootView.